python/lsst/obs/base/formatters/fitsExposure.py

```diff
@@ -85,201 +207,35 @@ class FitsImageFormatterBase(Formatter):
             variance: *default
 
     """
-    supportedExtensions = frozenset({".fits", ".fits.gz", ".fits.fz", ".fz", ".fit"})
-    extension = ".fits"
-    _metadata = None
     supportedWriteParameters = frozenset({"recipe"})
     _readerClass: type  # must be set by concrete subclasses
 
-    unsupportedParameters = {}
-    """Support all parameters."""
-
     @property
-    def metadata(self):
-        """The metadata read from this file. It will be stripped as
-        components are extracted from it
-        (`lsst.daf.base.PropertyList`).
-        """
-        if self._metadata is None:
-            self._metadata = self.readMetadata()
-        return self._metadata
-
-    def readMetadata(self):
-        """Read all header metadata directly into a PropertyList.
-
-        Returns
-        -------
-        metadata : `~lsst.daf.base.PropertyList`
-            Header metadata.
-        """
-        md = readMetadata(self.fileDescriptor.location.path)
-        fix_header(md)
-        return md
-
-    def stripMetadata(self):
-        """Remove metadata entries that are parsed into components.
-
-        This is only called when just the metadata is requested; stripping
-        entries there forces code that wants other components to ask for those
-        components directly rather than trying to extract them from the
-        metadata manually, which is fragile.  This behavior is an intentional
-        change from Gen2.
-
-        Parameters
-        ----------
-        metadata : `~lsst.daf.base.PropertyList`
-            Header metadata, to be modified in-place.
-        """
-        # TODO: make sure this covers everything, by delegating to something
-        # that doesn't yet exist in afw.image.ExposureInfo.
-        from lsst.afw.image import bboxFromMetadata
-        from lsst.afw.geom import makeSkyWcs
-
-        # Protect against the metadata being missing
-        try:
-            bboxFromMetadata(self.metadata)  # always strips
-        except LookupError:
-            pass
-        try:
-            makeSkyWcs(self.metadata, strip=True)
-        except Exception:
-            pass
-
-    def readComponent(self, component, parameters=None):
-        """Read a component held by the Exposure.
-
-        Parameters
-        ----------
-        component : `str`, optional
-            Component to read from the file.
-        parameters : `dict`, optional
-            If specified, a dictionary of slicing parameters that
-            overrides those in ``fileDescriptor``.
-
-        Returns
-        -------
-        obj : component-dependent
-            In-memory component object.
+    @cached_getter
+    def reader(self):
```

Comment:
Just because I don't know whether to put leading underscores on "protected" things.  If we do want leading underscores on this, I think they belong on every method here other than `read` and `write` and the handful of class attributes defined by `Formatter` itself; every other method should only ever be called by other methods in the same inheritance tree.  I lean _slightly_ towards no underscores on protected things _here_, just because I'd prefer not to have add obs_subaru and obs_decam branches for this ticket, but if we go that way, I should probably at least rename `_readerClass` -> `readerClass` (or `ReaderClass`) for consistency.  But I don't have a general preference or a strong one.